qt/scientific_interfaces/EnggDiffraction/EnggDiffGSASFittingViewQtWidget.cpp

```diff
@@ -0,0 +1,210 @@
+  const auto filenames(
+      QFileDialog::getOpenFileNames(this, tr("Find focused run files")));
+  setFocusedRunFileNames(filenames);
+}
+
+void EnggDiffGSASFittingViewQtWidget::displayLatticeParams(
+    const Mantid::API::ITableWorkspace_sptr latticeParams) const {
+  UNUSED_ARG(latticeParams);
+  throw std::runtime_error("displayLatticeParams not yet implemented");
+}
+
+void EnggDiffGSASFittingViewQtWidget::displayRwp(const double rwp) const {
+  UNUSED_ARG(rwp);
+  throw std::runtime_error("displayRwp not yet implemented");
+}
+
+std::vector<std::string>
+EnggDiffGSASFittingViewQtWidget::getFocusedFileNames() const {
+  const auto filenamesQStringList = m_ui.lineEdit_runFile->text().split(",");
+  std::vector<std::string> filenames;
+
+  for (const auto &filenameQString : filenamesQStringList) {
+    filenames.push_back(filenameQString.toStdString());
+  }
+  return filenames;
+}
+
+std::string EnggDiffGSASFittingViewQtWidget::getGSASIIProjectPath() const {
+  throw std::runtime_error("getGSASIIProjectPath not yet implemented");
+}
+
+std::string EnggDiffGSASFittingViewQtWidget::getInstrumentFileName() const {
+  throw std::runtime_error("getInstrumentFileName not yet implemented");
+}
+
+std::string EnggDiffGSASFittingViewQtWidget::getPathToGSASII() const {
+  throw std::runtime_error("getPathToGSASII not yet implemented");
+}
+
+double EnggDiffGSASFittingViewQtWidget::getPawleyDMin() const {
+  throw std::runtime_error("getPawleyDMin not yet implemented");
+}
+
+double EnggDiffGSASFittingViewQtWidget::getPawleyNegativeWeight() const {
+  throw std::runtime_error("getPawleyNegativeWeight not yet implemented");
+}
+
+std::vector<std::string>
+EnggDiffGSASFittingViewQtWidget::getPhaseFileNames() const {
+  throw std::runtime_error("getPhaseFileNames not yet implemented");
+}
+
+GSASRefinementMethod
+EnggDiffGSASFittingViewQtWidget::getRefinementMethod() const {
+  throw std::runtime_error("getRefinementMethod not yet implemented");
+}
+
+RunLabel EnggDiffGSASFittingViewQtWidget::getSelectedRunLabel() const {
+  const auto currentItemLabel =
+      m_ui.listWidget_runLabels->currentItem()->text();
+  const auto pieces = currentItemLabel.split("_");
+  if (pieces.size() != 2) {
+    throw std::runtime_error(
+        "Unexpected run label: \"" + currentItemLabel.toStdString() +
+        "\". Please contact the development team with this message");
+  }
+  return RunLabel(pieces[0].toInt(), pieces[1].toUInt());
+}
+
+void EnggDiffGSASFittingViewQtWidget::loadFocusedRun() {
+  m_presenter->notify(IEnggDiffGSASFittingPresenter::LoadRun);
+}
+
+void EnggDiffGSASFittingViewQtWidget::plotCurve(
+    const std::vector<boost::shared_ptr<QwtData>> &curves) {
+
+  for (size_t i = 0; i < curves.size(); ++i) {
+    auto *curve = curves[i].get();
+    auto plotCurve = Mantid::Kernel::make_unique<QwtPlotCurve>();
+
+    plotCurve->setData(*curve);
+    plotCurve->attach(m_ui.plotArea);
+    m_focusedRunCurves.push_back(std::move(plotCurve));
```

Comment:
We know how many curves we're going to create ahead of time, so we should use `m_focusedRunCurves.reserve(size)` outside the loop